# Reports
1. Inputs: analytics, metrics and business problem
2. Outputs: results and conclusions about business problems

# 1.0 Imports

###### 1.1 Libraries

In [1]:
import pandas as pd

###### 1.2 Colecting data

In [8]:
df = pd.read_csv('D:\\My Drive\\Pessoal\\Projetos\\bank_churn_predict\\bank_churn_predict\\churn.csv')

# 2.0 Baseline

###### 2.1 Média de churn atual

In [37]:
print(f'A taxa atual de churn é igual a: {round(train["Exited"].mean()*100,2)}%')

A taxa atual de churn é igual a: 20.37%


# 3.0 Métrica do Algorítmo Atual

# 4.0 Qual a taxa atual de Churn da Top-Bank, qual é o seu custo financeiro para a empresa por perda de faturamento?
1. A taxa atual de churn é igual a: 20.37%
2. O custo total para a empresa é igual a: $ 38,846,324.63

### 4.1 Premissas:

1. "Segundo o time de Analytics da TopBank, cada cliente que possui essa conta bancária retorna um valor monetário de 15% do valor do seu salário estimado, se esse for menor que a média e 20% se esse salário for maior que a média, durante o período vigente de sua conta. Esse valor é calculado anualmente."

### 4.2 Calculos:

###### 4.2.1 Média salarial

In [13]:
print(f'Média salarial igual a: ${round(df["EstimatedSalary"].mean(), 2)}')

Média salarial igual a: $100090.24


###### 4.2.1 15% da somatória dos salários estimados dentre os clientes em churn com salário estimado ABAIXO da média

In [27]:
clientes_abaixo_da_media = df.loc[(df["Exited"]==1)&(df["EstimatedSalary"]<100090.24), "EstimatedSalary"].sum()*.15
print(f'Soma total igual a: ${round(clientes_abaixo_da_media, 2)}')

Soma total igual a: $7472377.18


###### 4.2.2 20% da somatória dos salários estimados dentre os clientes em churn com salário estimado ACIMA da média

In [30]:
clientes_acima_da_media = df.loc[(df["Exited"]==1)&(df["EstimatedSalary"]>=100090.24), "EstimatedSalary"].sum()*.20
print(f'Soma total igual a: ${round(clientes_acima_da_media, 2)}')

Soma total igual a: $31373947.46


###### 4.2.3 Total

In [32]:
total = clientes_acima_da_media + clientes_abaixo_da_media

In [33]:
print(f'Soma total igual a: ${round(total, 2)}')

Soma total igual a: $38846324.63


# 5.0 Qual a performance do modelo em classificar os clientes como churn?